app/backend/app.py

```diff
@@ -110,14 +118,18 @@ async def ask():
                 request_json["messages"], context=context, session_state=request_json.get("session_state")
             )
         return jsonify(r)
-    except Exception as e:
-        logging.exception("Exception in /ask")
-        return jsonify({"error": str(e)}), 500
+    except Exception as error:
+        logging.exception("Exception in /ask: %s", error)
+        return jsonify(error_dict(error)), 500
 
 
 async def format_as_ndjson(r: AsyncGenerator[dict, None]) -> AsyncGenerator[str, None]:
-    async for event in r:
-        yield json.dumps(event, ensure_ascii=False) + "\n"
+    try:
```

Comment:
Unfortunately, our current try/catch wasn't catching exceptions thrown from inside the async generator. I posted a question in the Quart forum to try to understand why that is (https://github.com/pallets/quart/discussions/285) but for now, this approach works.
Also, interestingly, an error while streaming means a 200 still, since status code gets sent first, but then an {error} JSON in the stream.
Not sure if that's standard for streaming protocols.